src/main/java/de/tum/cit/aet/application/service/ApplicationService.java

```diff
@@ -46,58 +46,37 @@ public class ApplicationService {
      * Creates a new job application for the given applicant and job.
      * If an application already exists for the applicant and job, an exception is thrown.
      *
-     * @param createApplicationDTO DTO containing application and applicant data
+     * @param jobId the id of the job
+     * @param applicantId the id of the applicant
      * @return the created ApplicationForApplicantDTO
      * @throws OperationNotAllowedException if the applicant has already applied for the job
      */
     @Transactional
-    public ApplicationForApplicantDTO createApplication(CreateApplicationDTO createApplicationDTO) {
-        if (
-            applicationRepository.existsByApplicantUserIdAndJobJobId(
-                createApplicationDTO.applicant().user().userId(),
-                createApplicationDTO.jobId()
-            )
-        ) {
+    public ApplicationForApplicantDTO createApplication(UUID jobId, UUID applicantId) {
+        if (applicationRepository.existsByApplicantUserIdAndJobJobId(jobId, applicantId)) {
             throw new OperationNotAllowedException("Applicant has already applied for this position");
         }
-        Applicant applicant = applicantRepository.getReferenceById(UUID.fromString("00000000-0000-0000-0000-000000000104"));
-        applicant.setFirstName(createApplicationDTO.applicant().user().firstName());
-        applicant.setLastName(createApplicationDTO.applicant().user().lastName());
-        applicant.setGender(createApplicationDTO.applicant().user().gender());
-        applicant.setNationality(createApplicationDTO.applicant().user().nationality());
-        applicant.setBirthday(createApplicationDTO.applicant().user().birthday());
-        applicant.setPhoneNumber(createApplicationDTO.applicant().user().phoneNumber());
-        applicant.setWebsite(createApplicationDTO.applicant().user().website());
-        applicant.setLinkedinUrl(createApplicationDTO.applicant().user().linkedinUrl());
-        if (createApplicationDTO.applicant().user().selectedLanguage() != null) {
-            applicant.setSelectedLanguage(createApplicationDTO.applicant().user().selectedLanguage());
+        Optional<Applicant> applicantOptional = applicantRepository.findById(applicantId);
+        if (applicantOptional.isEmpty()) {
+            throw new IllegalArgumentException("The applicantId is not valid.");
```

Comment:
Is there a reason why you not throw an EntityNotFoundException here?

This part can also be simpliefied to:
```suggestion
            Applicant applicant = applicantRepository.findById(applicantId).orElseThrow(() -> new EntityNotFoundException(...));
```

Reply:
Thank you for your suggestion, you are right, that the exceptions fits much better, and your code suggestion looks much cleaner as well.
I implemented them like this